lib/active_model/serializer/configuration.rb

```diff
@@ -29,6 +29,28 @@ def config.array_serializer
         config.jsonapi_include_toplevel_object = false
         config.key_transform = nil
 
+        # example output
+        # => [
+        # "::Api::V1::ResNamespace::ResourceSerializer",
+        # "::Api::ResNamespace::ResourceSerializer",
+        # "::ResNamespace::ResourceSerializer" ,
+        # "::ResourceSerializer"]
+        config.serializer_lookup = Proc.new { |controller_name|
+          lookups = []
+          current_resource = controller_name.gsub(/Controller\z/, '')
+          current_resource += 'Serializer'
+          namespaces = current_resource.split('::')
+          resource_serializer = namespaces.slice!(-1)
+
+          lookups << '::' + resource_serializer
+          namespaces.length.times do |i|
+            current_namespace = namespaces[i - namespaces.length..-1].join('::')
+            lookups << '::' + current_namespace + '::' + resource_serializer
+          end
+
+          lookups
```

Comment:
This should be in serialization_context, maybe?.

It's also pretty easy to solve with explicit code in the controller.  then just pass in `serializer: resource_serializer` (or in index, `each_serializer: resource_serializer`) and you're good to go

```ruby
    def resource_serializer
      instance_variable_get("@#{resource_name}_serializer") || self.resource_serializer = "#{self.class.parent}::#{resource_class.name}Serializer".constantize
    end

    def resource_serializer=(new_resource_serializer)
      instance_variable_set("@#{resource_name}_serializer", new_resource_serializer)
    end

    # The resource class based on the controller
    # @return [Class]
    def resource_class
      @resource_class ||= resource_name.classify.constantize
    end

    # The singular name for the resource class based on the controller
    # @return [String]
    def resource_name
      @resource_name ||= controller_name.singularize
    end

```